projects/aca-content/viewer/src/lib/components/viewer/viewer.component.spec.ts

```diff
@@ -22,41 +22,332 @@
   let component: AcaViewerComponent;
+  let router: Router;
+  let route: ActivatedRoute;
+  let contentApi: ContentApiService;
+  let uploadService: UploadService;
   let nodesApiService: NodesApiService;
+  let appHookService: AppHookService;
   let store: Store<any>;
 
   beforeEach(() => {
     TestBed.configureTestingModule({
-      imports: [CoreTestingModule, StoreModule.forRoot({})]
+      imports: [
+        AcaViewerModule,
+        NoopAnimationsModule,
+        HttpClientModule,
+        RouterTestingModule.withRoutes([]),
+        TranslateModule.forRoot(),
+        StoreModule.forRoot(
+          { app: (state) => state },
+          {
+            initialState: {
+              app: INITIAL_APP_STATE
+            },
+            runtimeChecks: {
+              strictStateImmutability: false,
+              strictActionImmutability: false
+            }
+          }
+        ),
+        EffectsModule.forRoot([]),
+        PipeModule
+      ],
+      providers: [
+        { provide: AlfrescoApiService, useClass: AlfrescoApiServiceMock },
+        { provide: TranslationService, useClass: TranslationMock },
+        { provide: DocumentBasePageService, useVale: new DocumentBasePageServiceMock() },
+        {
+          provide: DiscoveryApiService,
+          useValue: {
+            ecmProductInfo$: new BehaviorSubject<RepositoryInfo | null>(null),
+            getEcmProductInfo: (): Observable<RepositoryInfo> =>
+              of(
+                new RepositoryInfo({
+                  version: {
+                    major: '10.0.0'
+                  } as VersionInfo
+                })
+              )
+          }
+        },
+        {
+          provide: AuthenticationService,
+          useValue: {
+            isEcmLoggedIn: (): boolean => true,
+            getRedirect: (): string | null => null,
+            setRedirect() {},
+            isOauth: (): boolean => false,
+            isOAuthWithoutSilentLogin: (): boolean => false
+          }
+        }
+      ]
     });
 
-    store = TestBed.inject(Store);
-    spyOn(store, 'select').and.returnValue(EMPTY);
     fixture = TestBed.createComponent(AcaViewerComponent);
     component = fixture.componentInstance;
+
+    router = TestBed.inject(Router);
+    route = TestBed.inject(ActivatedRoute);
+    contentApi = TestBed.inject(ContentApiService);
+    uploadService = TestBed.inject(UploadService);
     nodesApiService = TestBed.inject(NodesApiService);
+    appHookService = TestBed.inject(AppHookService);
+    store = TestBed.inject(Store);
+  });
+
+  it('should display document upon init', () => {
+    route.params = of({
+      folderId: 'folder1',
+      nodeId: 'node1'
+    });
+    spyOn(component, 'displayNode').and.stub();
+
+    component.ngOnInit();
+
+    expect(component.folderId).toBe('folder1');
+    expect(component.displayNode).toHaveBeenCalledWith('node1');
+  });
+
+  it('should not display node when id is missing', async () => {
```

Comment:
I would change a bit title because this test checks if getNodeInfo and navigate functions are not called. In reality it does not check if node is not displayed because there is no 'expect' which directly checks displaying of node